.github/workflows/main.yaml

```diff
@@ -189,6 +219,58 @@ jobs:
           type: ondevice
           os: linux
 
+  # Runs on-host integration and unit tests.
+  on-device-test:
+    needs: [initialize, build]
+    if: needs.initialize.outputs.on_device_test == 'true' && (contains(github.workflow, '_nightly') || contains(github.event.pull_request.labels.*.name, 'on_device'))
+    runs-on: [self-hosted, linux, X64]
+    name: On Device Test (${{ matrix.config }}/${{ matrix.type }} on ${{ matrix.platform }})
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: ${{ fromJson(needs.initialize.outputs.platforms) }}
+        config: [devel]
+        type: [unit_test, black_box_test]
```

Comment:
We also have 'evergreen' test, see https://cobalt-buildbot.corp.goog/builders/raspi-2_evergreen-arm-hardfp_evergreen_tests/builds/276.  We should probably move this to respected platform configuration files.

Reply:
Resolved